File: recipes/replace_string/go/replace_string.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	// Target string
	str := "dog cat dog"

	// Show original string
	fmt.Println(str)

	// Replace one dog to panda
	fmt.Println(strings.Replace(str, "dog", "panda", 1))

	// Replace two dogs to panda
	fmt.Println(strings.Replace(str, "dog", "panda", 2))

	// Replace no dog
	fmt.Println(strings.Replace(str, "dog", "panda", 0))

	// Replace dog to panda infinitely
	fmt.Println(strings.Replace(str, "dog", "panda", -1))
}

File: recipes/replace_string/php/replace_string.php
<?php
// Replace "Super" to "Great"
$str_original  = "Super panda";
$str_replaced1 = str_replace("Super", "Great", $str_original);
echo $str_replaced1 . "\n";
echo "\n";

// Replace multiple words using array
$search_array  = array("foo", "bar");
$str_replaced2 = str_replace($search_array, "panda", "foo, bar");
echo $str_replaced2 . "\n";
echo "\n";

// Get the number of replaced strings
$count = 0;
$str_replaced3 = str_replace("dog", "panda", "dog, dog, dog!", $count);
echo $str_replaced3 . "\n";
echo $count . " dogs are replaced to panda\n";
?>